### PR TITLE
BUG: Fix return of missing values when applying loc to single level of MultiIndex

### LINKED ISSUE #27104: Partial indexing with list on MultiIndex with missing value includes them despite not being in the list
#### Code Sample, a copy-pastable example if possible

```python
In [2]: df = pd.DataFrame({'col':[1, 2, 3, 4, 5], 'ind1':['a','b','c','d',np.nan], 'ind2':[1,2,3,4,5] }).set_index(['ind1', 'ind2'])                                                                                                          

In [3]: df.loc['a']                                                                                                                                                                                                                           
Out[3]: 
      col
ind2     
1       1

In [4]: df.loc[['a']]                                                                                                                                                                                                                         
Out[4]: 
           col
ind1 ind2     
a    1       1
NaN  5       5


```
#### Problem description

``Out[3]`` is correct, ``Out[4]`` is wrong - the ``NaN`` line should not be there.

From #15107.

#### Expected Output

Same as ``Out[3]``.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 45ea26763da832189747ac9f86630fece84b4f18
python           : 3.7.3.candidate.1
python-bits      : 64
OS               : Linux
OS-release       : 4.9.0-9-amd64
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : it_IT.UTF-8
LOCALE           : it_IT.UTF-8

pandas           : 0.25.0.dev0+824.g45ea26763
numpy            : 1.16.4
pytz             : 2016.7
dateutil         : 2.8.0
pip              : 9.0.1
setuptools       : 41.0.1
Cython           : 0.29.2
pytest           : 3.0.6
hypothesis       : 3.6.1
sphinx           : 1.4.9
blosc            : None
feather          : None
xlsxwriter       : 0.9.6
lxml.etree       : 4.3.2
html5lib         : 0.999999999
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: 0.2.1
bs4              : 4.5.3
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.2
matplotlib       : 3.0.2
numexpr          : 2.6.9
openpyxl         : 2.3.0
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.1.0
sqlalchemy       : 1.0.15
tables           : 3.4.4
xarray           : None
xlrd             : 1.0.0
xlwt             : 1.3.0
xlsxwriter       : 0.9.6


</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #27104
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This confused ``-1`` representing missing values in input with the ``-1`` signaling which components to exclude from output.

### CODE REVIEW

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 1]
```diff
@@ -598,3 +598,15 @@ def test_getitem_loc_commutability(multiindex_year_month_day_dataframe_random_da
     result = ser[2000, 5]
     expected = df.loc[2000, 5]["A"]
     tm.assert_series_equal(result, expected)
+
+
+def test_loc_with_nan():
+    # GH: 27104
+    df = DataFrame(
+        {"col": [1, 2, 5], "ind1": ["a", "d", np.nan], "ind2": [1, 4, 5]}
+    ).set_index(["ind1", "ind2"])
```
Comment (jreback) [inline pandas/tests/indexing/multiindex/test_loc.py:607]: can you also select `df.loc['a']` as the first case in OP (which is working). can parameterize if easy.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. ping on green.

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexing/multiindex/test_loc.py:607]: Parametrization would be ugly, because Index needs new definition. Added the test below.

Comment (jreback) [reply pandas/tests/indexing/multiindex/test_loc.py:607]: kk totally fine, i always ask, but like you did push back if its non-trivial or would make it less obvious.

**Review by jreback — APPROVED** [review]

**Comment by phofl** [comment]
cc @jreback green


Thanks for merging master

File: doc/source/whatsnew/v1.2.0.rst [hunk 2]
```diff
@@ -466,7 +466,8 @@ Indexing
 - Bug in indexing on a :class:`Series` or :class:`DataFrame` with a :class:`MultiIndex` with a level named "0" (:issue:`37194`)
 - Bug in :meth:`Series.__getitem__` when using an unsigned integer array as an indexer giving incorrect results or segfaulting instead of raising ``KeyError`` (:issue:`37218`)
 - Bug in :meth:`Index.where` incorrectly casting numeric values to strings (:issue:`37591`)
-- Bug in :meth:`Series.loc` and :meth:`DataFrame.loc` raises when numeric label was given for object :class:`Index` although label was in :class:`Index` (:issue:`26491`)
+- Bug in :meth:`DataFrame.loc` returned requested key plus missing values when ``loc`` was applied to single level from :class:`MultiIndex` (:issue:`27104`)
+- Bug in :meth:`Series.loc` and :meth:`DataFrame.loc` raises when numeric label was given for object :class:`Index` although label was in :class:`Index` (:issue:`26491`)>>>>>>> master
```
Comment (charlesdong1991) [inline doc/source/whatsnew/v1.2.0.rst:470]: seems there is a `>>>>>>> master`?

Comment (phofl) [reply doc/source/whatsnew/v1.2.0.rst]: Looks like merge went wrong. Is fixed now. Thanks

**Comment by charlesdong1991** [comment]
very nice @phofl

**Pull request merged**